source/Mlos.Python/mlos/Optimizers/OptimizerBase.py

```diff
@@ -56,15 +56,31 @@ def predict(self, feature_values_pandas_frame, t=None) -> Prediction:
         """
         raise NotImplementedError("All subclasses must implement this method.")
 
-    @abstractmethod
-    def optimum(self, stay_focused=False) -> Dict: # TODO: make it return an object
+    def optimum(self, stay_focused=False) -> Dict: # pylint: disable=unused-argument  # TODO take context
         """ Return the optimal value found so far along with the related parameter values.
 
         This could be either min or max, depending on the settings.
 
-        :return:
+        Returns
+        -------
+        best_config_point : Point
+            Configuration that corresponds to the optimum objective value.
+        best_objective_point : Point
+
+            Best objective value observed so far (corresponding to the best_config_point).
+
         """
-        raise NotImplementedError("All subclasses must implement this method.")
+        features_df, objectives_df = self.get_all_observations()

```

Comment:
What if the user calls optimum before registering any observations?

Reply:
It errors awkwardly right now. I can make it throw a nicer error?

Reply:
throwing a nicer error now.